app/views/users/show.html.erb

```diff
@@ -152,6 +152,13 @@
       </button>
       <div id="user-profile-dropdownmenu" class="profile-dropdownmenu">
         <button id="user-profile-dropdownmenu-block-button" data-profile-user-id="<%= @user.id %>" class="block-button" type="button">Block @<%= @user.username %></button>
+        <% if current_user&.has_role?(:trusted) %>
+          <%= button_to Reaction.cached_any_reactions_for?(@user, current_user, "vomit") ? "Flag as Abusive" : "Remove Abuse Flag",
+                        reactions_path(reactable_id: @user.id, category: "vomit", reactable_type: "User"),
+                        method: :post,
+                        remote: true,
+                        form_class: "block-button" %>
+        <% end %>
```

Comment:
I think for this specific case we could make it not `remote: true` or put the JS logic in Webpacker and have it `import` a new file in that `app/javascript/profileDropdown`.

Reply:
`<% if current_user&.has_role?(:trusted) %>`

Does this mean that the button would only show on tag/site mods and the core team?

Reply:
@reyesvicente yep, because this feature will have impact on a person's articles, we're allowing just mods (and higher) to use this.

Also @jacobherrington we actually don't have `current_user` available here because of caching, so we'll probably want to render this all the time with `display: none` and then use JS to set it `display: block` or something.